avm/res/insights/data-collection-rule/modules/nested_lock.bicep

```diff
@@ -0,0 +1,32 @@
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Name of the Data Collection Rule to apply the lock to.')
+param dataCollectionRuleName string
+
+resource dataCollectionRule 'Microsoft.Insights/dataCollectionRules@2023-03-11' existing = {
+  name: dataCollectionRuleName
+}
+
+resource dataCollectionRule_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${dataCollectionRuleName}'
+  scope: dataCollectionRule
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete'
+      ? 'Cannot delete resource or child resources.'
+      : 'Cannot delete or modify the resource or child resources.'
+  }
+}
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type lockType = {
```

Comment:
If youi define the type here too, I'd advice to add an `export()` flag and import it in the parent. Then you have one less place to maintain it from.

Reply:
addressed in: https://github.com/Azure/bicep-registry-modules/pull/3211/commits/dd9ddf789ccb41bf73fa82992e84dfeb59ed51bd